src/cache.cpp

```diff
@@ -248,7 +248,7 @@ bool StatCache::GetStat(const std::string& key, struct stat* pst, headers_t* met
                     std::string tag = lower(hiter->first);
                     if(tag == "etag"){
                         stretag = hiter->second;
-                        if('\0' != petag[0] && 0 != strcmp(petag, stretag.c_str())){
+                        if('\0' != petag[0] && petag != stretag){
```

Comment:
By the way, why don't you make the following changes?
`'\0' != petag[0]` -> `!petag.empty()`

Reply:
`petag` is a `char*`.

Reply:
Sorry for my fault.